sktime/forecasting/ensemble_algorithms.py

```diff
@@ -0,0 +1,234 @@
+from sklearn.base import BaseEstimator
+from scipy.optimize import nnls
+import numpy as np
+
+
+class EnsembleAlgorithm(BaseEstimator):
+    """Wrapper class to take ensemble algorithms and allow them to be easily set
+
+    Parameters
+    ----------
+    n : float
+    loss_func : function
+    """
+    def __init__(self, n=10, loss_func=None, **kwargs):
+        self.n = n
+        self.weights = np.ones(n)/n
```

Comment:
could be replaced with call to `self._uniform_weights`, no? Given scikit-learn's and our API, I'd be careful to do variable manipulations inside init though, ideally move them to fit, to make sure that cloning and parameter setting works as expected, e.g. when I use `self.set_params(...)`, `__init__` wouldn't be called again but as a user I'd expect it to work

Reply:
I guess for `ensemble_algorithms`, I don't have a `fit` function. I was hoping that setting the uniform weights as a default would make sense to forgo a fit function, but I can also make it standard to first call `self._uniform_weights` before further updates. Would that work as a replacement? Additionally `OnlineEnsembleForecaster` uses something similar; should that behaviour be adjusted as well?